### PR TITLE
add HOST_IP to mesh-gateway

### PR DESCRIPTION
**Pull request by woz5999**
Changes proposed in this PR:
- Add `HOST_IP` environment variable back to mesh-gateway deployment
Expansion of this var is supported according to the docs, but requires the var to actually be set: https://developer.hashicorp.com/consul/docs/connect/proxies/envoy#envoy_statsd_url

How I've tested this PR:
- Applied the change locally and verified that the pod manifest reflect the updated environment configuration

How I expect reviewers to test this PR:
- Apply the changes are verify that the pod manifests reflect the updated environment configuration

Note too sure the best way to add a test for this. Suggestions would be greatly appreciated.

Checklist:
- [ ] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Comment by woz5999** [comment]
@kschoche anything i can to help move this forward?

**Comment by woz5999** [comment]
@t-eckert anything i can do to help get this reviewed?

**Comment by t-eckert** [comment]
Hi @woz5999, 

Thank you for reaching out and doing this work!

To get this reviewed, let's make just a few changes. Can you revert changes to the `CHANGELOG`? We will take care of updating it and there are some changes that look to be created by an automatic formatter.

I will tag myself as a reviewer and another engineer on Consul-K8s.

I'll test this out myself and we will get back to you.

Thank you again for opening this PR!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
 ## UNRELEASED
 
 IMPROVEMENTS:
+
 * Helm:
   * Add a `global.extraLabels` stanza to allow setting global Kubernetes labels for all components deployed by the `consul-k8s` Helm chart. [[GH-1778](https://github.com/hashicorp/consul-k8s/pull/1778)]
   * Add the `accessLogs` field to the `ProxyDefaults` CRD. [[GH-1816](https://github.com/hashicorp/consul-k8s/pull/1816)]
   * Add the `envoyExtensions` field to the `ProxyDefaults` and `ServiceDefaults` CRD. [[GH-1823]](https://github.com/hashicorp/consul-k8s/pull/1823)
   * Add the `balanceInboundConnections` field to the `ServiceDefaults` CRD. [[GH-1823]](https://github.com/hashicorp/consul-k8s/pull/1823)
+  * Add `HOST_IP` environment variable to mesh-gateway deployment
```
Comment (t-eckert) [inline CHANGELOG.md:10]: Thank you for adding this, but let's remove it from this PR before it gets reviewed. We'll add it into the CHANGELOG later.

Comment (woz5999) [reply CHANGELOG.md]: Removed

**Comment by woz5999** [comment]
Removed the changelog formatting changes. good catch.

**Review by t-eckert — APPROVED** [review]
Looks good to me! Thank you for catching it.

**Comment by david-yu** [comment]
Thanks @t-eckert. Adding backport labels as well. Not sure if this will make it in to this weeks release.

**Comment by david-yu** [comment]
@woz5999 just saw your issue on https://github.com/hashicorp/consul-dataplane/issues/62. It looks like consul-dataplane also needs to support using this variable is that correct?

**Comment by woz5999** [comment]
@david-yu yes, there are additional changes required in the mesh-gateway to support variable expansion in order to utilize this once it's added back.

**Comment by woz5999** [comment]
Will this go out in conjunction with the upcoming dataplane release?

**Comment by woz5999** [comment]
No worries. changelog added

**Comment by DanStough** [comment]
Tests are failing because this is from a fork and the changelog checker script has an error that will be closed in #2127. I did a quick e2e test with the dataplane changes and everything looks good, so I'm fine merging this.

**Pull request merged**